WooCommerce/WooCommerceTests/ViewRelated/Survey/SurveyViewControllerTests.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import WebKit
+import XCTest
+
+@testable import WooCommerce
+
+/// Test cases for `SurveyViewController`.
+///
+final class SurveyViewControllerTests: XCTestCase {
+
+    func testItLoadsTheCorrectInAppFeedbackSurvey() throws {
+        // Given
+        let viewController = SurveyViewController(survey: .inAppFeedback, onCompletion: {})
+
+        // When
+        _ = try XCTUnwrap(viewController.view)
+        let mirror = try self.mirror(of: viewController)
+
+        // Then
+        XCTAssertTrue(mirror.webView.isLoading)
+        XCTAssertEqual(mirror.webView.url, WooConstants.inAppFeedbackURL)
+    }
+
+    func testItCompletesAfterReceivingAFormSubmittedPostCallbackRequest() throws {
+        // Given
+        let exp = expectation(description: #function)
+        var surveyCompleted = false
+        let viewController = SurveyViewController(survey: .inAppFeedback, onCompletion: {
+            surveyCompleted = true
+            exp.fulfill()
+        })
+
+        // When
+        _ = try XCTUnwrap(viewController.view)
+        let mirror = try self.mirror(of: viewController)
+
+        // Fakes a form submission POST navigation
+        let navigationAction = FormSubmittedNavigationAction(httpMethod: "POST")
+        viewController.webView(mirror.webView, decidePolicyFor: navigationAction, decisionHandler: { _ in })
```

Comment:
I couldn't find a cleaner way to fake a form submission :-/
Let me know if you have better ideas 😛

Reply:
wow I didn't even know how to test this, `FormSubmittedNavigationAction` looks really cool!